### PR TITLE
Add ConvertSubtractWithConstant to MOCTransformations

### PR DESCRIPTION
**Pull request by mateusztabaka**
In addition to that ConvertSubtract/ConvertSubtractWithConstant was slightly
simplified thanks to changes in DisableConvertConstantFoldingOnConstPath.
DisableConvertConstantFoldingOnConstPath was renamed to MarkDequantizationSubgraph
and modified such that it also marks Subtract and Multiply from dequantization subgraph
with 'DequantizationNode' attribute. So we can easily skip ConvertSubtract for such nodes.

Tickets: 62419 and partially 65489

### CODE REVIEW

**Review by GlebKazantaev — APPROVED** [review]
In general changes looks good! Let's wait for validation cycles.

**Comment by GlebKazantaev** [comment]
@eshoguli  please take a look.

**Review by eshoguli — COMMENTED** [review]
In generally it's OK for me, but: can you, please, share DLB results before merge? We need to check possible performance degradations and load time.

minor, can be merged as is: `Dequantization` is simplier and shorter. We already had `Dequantization` attribute few years ago but it has been removed. Looks like we are going to return it here : )

File: src/common/transformations/include/transformations/mark_dequantization_subgraph.hpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ngraph/pass/graph_rewrite.hpp>
+#include <transformations_visibility.hpp>
+
+namespace ngraph {
```
Comment (praasz) [inline src/common/transformations/include/transformations/mark_dequantization_subgraph.hpp:10]: Is `ngraph` namespace intentional or can be changed to `ov`?

File: src/common/transformations/include/transformations/mark_dequantization_subgraph.hpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ngraph/pass/graph_rewrite.hpp>
```
Comment (praasz) [inline src/common/transformations/include/transformations/mark_dequantization_subgraph.hpp:7]: Use `""` instead `<>` as in other newly added files?
Can `openvino` includes be used?

File: src/common/transformations/src/transformations/mark_dequantization_subgraph.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                // so we don't have to constantfold it and then convert it back to
+                // low precision in LP transformations
+                ov::disable_constant_folding(zero_point->shared_from_this());
+            }
+        }
+
+        // mark Multiply as dequantization node
+        ov::mark_as_dequantization_node(m.get_match_root());
+
+        return false;
```
Comment (praasz) [inline src/common/transformations/src/transformations/mark_dequantization_subgraph.cpp:70]: Why always return `false` in callback?

File: src/common/transformations/src/transformations/op_conversions/convert_subtract.cpp [hunk 4]
```diff
@@ -4,75 +4,72 @@
...
+    }
+
+    if (!sub->get_input_element_type(1).is_signed()) {
+        return false;
+    }
+
+    std::shared_ptr<Node> neg =
+        std::make_shared<opset1::Multiply>(sub->input_value(1),
+                                           opset1::Constant::create(sub->get_input_element_type(1), Shape{}, {-1}));
+    NodeVector new_nodes;
```
Comment (praasz) [inline src/common/transformations/src/transformations/op_conversions/convert_subtract.cpp:34]: Consider use [NodeRegistry](https://github.com/openvinotoolkit/openvino/blob/f6d6f5629fa5679717de31e7a8a8506a77550ed0/src/core/include/openvino/pass/graph_rewrite.hpp#L22) class here.
Then you can make/add nodes by this registry and copy `runtime_info` as 
```cpp
 copy_runtime_info(sub, new_nodes.get());
```

File: src/common/transformations/src/transformations/op_conversions/convert_subtract.cpp [hunk 5]
```diff
@@ -4,75 +4,72 @@
...
 
 #include "itt.hpp"
+#include "transformations/rt_info/dequantization_node.hpp"
 
-ngraph::pass::ConvertSubtract::ConvertSubtract() {
+using namespace ngraph;
+
+static bool convert_subtract(const std::shared_ptr<Node>& node) {
+    auto sub = std::dynamic_pointer_cast<opset1::Subtract>(node);
+    if (!sub) {
```
Comment (praasz) [inline src/common/transformations/src/transformations/op_conversions/convert_subtract.cpp:19]: Is it required?
The `pattern::wrap_type<opset1::Subtract>` is not checking it?

File: src/tests/functional/inference_engine/transformations/op_conversions/convert_subtract.cpp [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        auto neg = std::make_shared<opset8::Multiply>(data2, opset8::Constant::create(element::f32, Shape{}, {-1}));
+        auto add2 = std::make_shared<opset8::Add>(data1, neg);
+
+        function_ref = std::make_shared<Function>(NodeVector{add1, add2}, ParameterVector{data1, data2});
+    }
+    comparator.enable(FunctionsComparator::CmpValues::CONST_VALUES);
+    comparator.enable(FunctionsComparator::CmpValues::ACCURACY);
+}
+
+TEST_F(TransformationTestsF, ConvertSubtractWithConstant) {
```
Comment (praasz) [inline src/tests/functional/inference_engine/transformations/op_conversions/convert_subtract.cpp:43]: Consider add some helper fixture:
```cpp
class TransformationConvertSubtractTest : public TransformationTestsF{};
```
and then convert test cases to e.g
```cpp
TEST_F(TransformationConvertSubtractTest , with_constant)
```

**Pull request closed without merging**